src/clima_sensor_us.rs

```diff
@@ -0,0 +1,523 @@
+/// Before using this module you need to configure elektra with the following elements
+/// [opensensemap/box_id] and [opensensemap/access_token]
+/// For example:
+/// kdb set user:/sw/libelektra/opensesame/#0/current/opensensemap/box_id "<opensensemap-box-id>"
+/// kdb set user:/sw/libelektra/opensesame/#0/current/opensensemap/access_token "<access-token>"
+extern crate libmodbus;
+
+use crate::config::Config;
+use libmodbus::*;
+use reqwest::header::HeaderMap;
+use reqwest::Client;
+
+///Constants
+const DEVICE: &'static str = "/dev/ttyS5";
+const BAUDRATE: i32 = 9600;
+const PARITY: char = 'N';
+const DATA_BITS: i32 = 8;
+const STOP_BITS: i32 = 1;
+const SLAVE_ID: u8 = 1;
+
+const ERROR_CODE_S32: u32 = 0x7FFFFFFF;
+const ERROR_CODE_U32: u32 = 0xFFFFFFFF;
+
+//Adresses of registers
+const REG_MEAN_WIND_SPEED: u16 = 0x88B9;
+const REG_MEAN_WIND_DIREC: u16 = 0x88BB;
+const REG_AIR_TEMP: u16 = 0x88BD;
+const REG_FRAME_TEMP: u16 = 0x88BF;
+const REG_ACOUSTIC_TEMP: u16 = 0x88C1;
+const REG_AIR_TEMP_UNCORRECTED: u16 = 0x88C3;
+const REG_REL_HUMIDITY: u16 = 0x88C5;
+const REG_DEW_POINT_TEMP: u16 = 0x88C7;
+const REG_ABS_AIR_PRESSURE: u16 = 0x88C9;
+const REG_REL_AIR_PRESSURE: u16 = 0x88CB;
+const REG_BRIGHTNESS_N: u16 = 0x88CD;
+const REG_BRIGHTNESS_E: u16 = 0x88CF;
+const REG_BRIGHTNESS_S: u16 = 0x88D1;
+const REG_BRIGHTNESS_W: u16 = 0x88D3;
+const REG_DIREC_BRIGHTNESS: u16 = 0x88D5;
+const REG_BRIGHTNESS_MAX: u16 = 0x88D7;
+const REG_PRECIPITATION_EVENT: u16 = 0x88D9;
+const REG_PRECIPITATION_INTENSITY: u16 = 0x88DB;
+const REG_PRECIPITATION_AMOUNT: u16 = 0x88DD;
+const REG_PRECIPITATION_TYPE: u16 = 0x88DF;
+const REG_SUN_ELEVATION: u16 = 0x88E9;
+const REG_SUN_AZIMUTH: u16 = 0x88EB;
+const REG_HEIGHT_ABOVE_SEA: u16 = 0x88ED;
+const REG_SENSOR_SUPPLY: u16 = 0x88F1;
+const REG_MAX_WIND_SPEED: u16 = 0x8901;
+const REG_WIND_DIREC: u16 = 0x8903;
+const REG_ABS_HUMIDITY: u16 = 0x8905;
+const REG_REL_HUMIDITY_UNCORRECTED: u16 = 0x8907;
+const REG_MAGNETIC_COMPASS_DIFF_ANGLE: u16 = 0x8909;
+const REG_BRIGHTNESS_VEC_SUM: u16 = 0x890B;
+const REG_WINDCHILL_TEMP: u16 = 0x890D;
+const REG_HEAT_INDEX_TEMP: u16 = 0x890F;
+const REG_ABS_PRECIPITATION_AMOUNT: u16 = 0x8911;
+const REG_GLOBAL_RADIATION: u16 = 0x8913;
+const REG_PITCH_MAGNETIC_COMPASS_NS: u16 = 0x8915;
+const REG_ROLL_MAGNETIC_COMPASS_EW: u16 = 0x8917;
+
+//OpenSenseMap
+const UPDATE_FREQU: u32 = 30000; // 30000ms*10ms = 5min
```

Comment:
Is it written like this in the data sheet? If it is, you can keep it as-is, otherwise I would prefer fully-written names:

```suggestion
const UPDATE_FREQUENCY: u32 = 30000; // 30000ms*10ms = 5min
```

Reply:
No, I used `UPDATE_FREQUENCY` to read all registered every 5 min. Instead of calling them all 10 ms. So it is nothing form the data sheet.

Reply:
And we probably should use 1 minute, as the web-interface from opensensemaps suggests that updates happen every minute.

Reply:
One important change needed: the weatherstation needs to be integrated in the sensors loop, so that messstation can get both weatherstation and air quality data. The frequency also needs to be adapted to make this work. (The sensors loop gets called much less frequently).

(Later we can remove all loops and use tokio instead but for now let us keep these two loops.)

Reply:
Is the sensor loop, where `environment.handle()` is called?

Reply:
yes, e.g. line 236, and writeln in 240 could be modified to log some of the weather data also on the HD.

Reply:
But for me, it looks like line 236 isn't called in an endless loop. I thought this is the initialization of the env sensor.

Reply:
Okay, so, in that case, the while loop (main.rs line 299) is never called and the other functions are implied disabled.

Reply:
Yes, in the sensors mode everything else is currently disabled.